modules/dex/src/lib.rs

```diff
@@ -1176,6 +1174,17 @@ impl<T: Config> Pallet<T> {
 		Ok(supply_amounts)
 	}
 
+	fn validate_path(path: &[CurrencyId]) -> DispatchResult {
+		let path_length = path.len();
+		ensure!(
+			path_length >= 2 && path_length <= T::TradingPathLimit::get().saturated_into(),
+			Error::<T>::InvalidTradingPathLength
+		);
+		ensure!(path[0] != path[path_length - 1], Error::<T>::InvalidTradingPath);
```

Comment:
```suggestion
		ensure!(path.get(0) != path.get(path_length - 1), Error::<T>::InvalidTradingPath);
```

`vector[n]` could panic, I think we should use `get` instead

Reply:
we checked length >= 2 so it shouldn't panic, but yeah can still improve it